Fix initially selected items in SelectControl component

joshuatf opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

* Allows selection of single select items and shows the value in the input
* Syncs downshift's internal state of selected items
* Adds stories with selected items

### How to test the changes in this Pull Request:

1. Run `pnpm --filter=@woocommerce/storybook storybook`
2. Navigate to `experimental / SelectControl`
3. View the multiple and single examples to check default selected values

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

joshuatf committed:
Allow initial selection of items in SelectControl

joshuatf committed:
Update stories to include initially selected items

joshuatf committed:
Add changelog entry

louwie17 commented on packages/js/components/src/experimental-select-control/select-control.tsx:
```diff
@@ -112,8 +120,9 @@ function SelectControl< ItemType = DefaultItemType >( {
 		getItemProps,
 		selectItem,
-		selectedItem: singleSelectedItem,
+		selectedItem: selectedItemSingle,
```
It confused me a bit what `selectedItemSingle` was referring to versus `singleSelectedItem`, given they are mostly just rearranging the same words :)
Could we change to to `comboboxSingleSelectedItem` or something like that, so we can clearly depict this is from `combobox` internal state?

joshuatf replied:
Definitely!  I like this name much better 😄

louwie17 commented on packages/js/components/src/experimental-select-control/select-control.tsx:
```diff
@@ -84,24 +82,34 @@ function SelectControl< ItemType = DefaultItemType >( {
+			return;
+		}
+
+		setInputValue( getItemLabel( singleSelectedItem ) );
```
Do you think we should add an extra check around this if the `selectedItem` is not already equal to the `singleSelectedItem`? I always forget if React checks that internally or not, or will still trigger a re-render despite nothing changing.
Scratch that, looks like React won't trigger a re-render if the label is the same: https://dev.to/sunflower/reactjs-if-it-is-setting-a-state-with-the-same-value-will-the-component-be-re-rendered-5g24

louwie17 reviewed (commented):
Thanks for fixing this @joshuatf, I just left one minor comment in terms of a variable name, outside of that this looks great :)

joshuatf committed:
Update combobox single selected item name

joshuatf commented:
Thanks for the feedback, @louwie17!  Addressed your feedback around the name in https://github.com/woocommerce/woocommerce/pull/34794/commits/4c7571025a30cfe6a4f49ca8663a0138921387f7 and this is ready for another look.

louwie17 reviewed (dismissed):
Thanks for updating the variable name, this LGTM 🚀

joelclimbsthings reviewed (approved):
Looks good @joshuatf ! 🚢
